### ISSUE TITLE
Open files in RStudio from Terminal pane

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#14226]
Some 3.5 year after the last post I landed here for similar question.

I am working remotely on data and using Rstudio server, and there are files that are not in the working directory and the fastest way is to use terminal and navigate there. For instance one can use `fzf` which nicely works inside Rstudio terminal to fuzzy-find the file they want. It would be ideal if it was possible to open a file from terminal in the rstudio server. The current workflow:
1. navigate to the path in terminal
2. use `fzf`, `find`, `fd`, `ls` to find the file
3. use the mouse to copy the path
4. go to Rstudio console
5. type `file.edit("`
6. paste the address and press enter

Ideal workflow number 1:
1. navigate to the path in terminal
2. use `fzf`, `find`, `fd`, `ls` to find the file and pipe it to some shell function to open it in Rstudio

Ideal workflow number 2:
1. navigate to the path in terminal
2. use `ls` to find the file and click on it to open it in rstudio

_Originally posted by @mmahmoudian in https://github.com/rstudio/rstudio/issues/1850#issuecomment-1928284978_

**Comment by gtritchie** [comment]
Also mentioned here: https://github.com/rstudio/rstudio/issues/2377

**Comment by gtritchie** [comment]
I started working on this back in 2018, but other priorities won out. Don't recall what approach I was taking, but could probably rediscover it if we decide to proceed with this.

**Comment by julianbarg** [comment]
I still kinda want this. Opening files from an external terminal does not work.

**Comment by gtritchie** [comment]
I'll take another look for the next release; no promises but I'd also really like to implement this!

**Comment by gtritchie** [comment]
There is a partial workaround for this for RStudio Desktop (but not Server or Workbench). I'll still be looking into a more robust solution that will work the same on all platforms, but just FYI.

If you've got RStudio registered as the handler for a file type (e.g. `.R` files) such that double-clicking on such a file in the system's file explorer opens it in RStudio, then the following in the RStudio terminal will (should) work, but probably somewhat unpredictably if you've got multiple instances of RStudio open.

## macOS
```bash
open foo.R
```

## Linux
```bash
xdg-open foo.R
```

## Windows (with a Command prompt or PowerShell terminal)
```cmd
start foo.R
```

**Comment by julianbarg** [comment]
Yes, can confirm that `xdg-open foo.R` works in this fashion, with the caveat you mentioned. I must have only tried `rstudio foo.R`. Being able to open a file in a specific project would be a nice quality of life improvement. Although you can have multiple instances of the same project open at the same time. Thinking of it, other editors don't seem to handle this gratefully either. Sublime Text pretty much never gets it right.

**Comment by gtritchie** [comment]
Current milestone is already too full (we're switching from quarterly releases to monthly releases for RStudio), bumping to the next release.

**Comment by katrinabrock** [comment]
Thank you!! This is something I've wanted to be able to do for quite a while.

**Comment by kevinushey** [comment]
Sorry it took so long! You should see it as part of the upcoming RStudio release, currently planned for October. You can also give it a test drive in the daily builds at https://dailies.rstudio.com/. Thanks!
